components/Components/Query.php

```diff
@@ -245,30 +246,40 @@ public function withSeparator(string $separator): self
 
     public function sort(): self
     {
-        if (count($this->pairs) === count(array_count_values(array_column($this->pairs, 0)))) {
-            return $this;
+        $parameters = array_reduce($this->pairs, function (array $carry, array $pair) {
+            $carry[$pair[0]] ??= [];
+            $carry[$pair[0]][] = $pair[1];
+
+            return $carry;
+        }, []);
+
+        $codepoints = fn (?string $str): string => in_array($str, ['', null], true) ? '' : implode('.', array_map(
+            mb_ord(...), /* @phpstan-ignore-line */
+            (array) preg_split(pattern:'//u', subject: $str, flags: PREG_SPLIT_NO_EMPTY)
+        ));
```

Comment:
Good idea! 😁

Reply:
thanks took me some thinking to come up with this